### PR TITLE
Update project_charter.md: clarify maintenance of older releases

### PR DESCRIPTION
**Pull request by climbfuji**
### Summary

Following today's discussion in the spack-stack developer meeting

### Testing

n/a

### Applications affected

n/a

### Systems affected

n/a

### Dependencies

n/a

### Issue(s) addressed

n/a

### Checklist
- [x] This PR addresses one issue/problem/enhancement, or has a very good reason for not doing so.
- [ ] ~~These changes have been tested on the affected systems and applications.~~
- [ ] ~~All dependency PRs/issues have been resolved and this PR can be merged.~~

### CODE REVIEW

**Comment by climbfuji** [comment]
@AlexanderRichert-NOAA How about this?

**Comment by AlexanderRichert-NOAA** [comment]
What about an additional sentence: "Modifications to past releases that do not impact reinstallation, such as the addition of chained environments/custom Spack repositories, or new templates or site configurations, may be considered on a case-by-case basis."? Then in practice, we put as much of the development burden on the requesting developers as possible.

Also, referts->refers

File: project_charter.md [hunk 1]
```diff
@@ -189,6 +189,6 @@ Since official updates require coordination across all platforms, the timeline f
 ## Lifetime of spack-stack installations
 Regular, full releases of spack-stack are made every three months around the end of February, May, August, and November.
 
-spack-stack developers are committed to maintaining a rolling stock of the last four releases, i.e. one year's worth of spack-stack releases. Under special circumstances (e.g. operational implementations that require long-term support, long-running official retrospectives), selected releases may be kept longer.
+spack-stack developers are committed to maintaining a rolling stock of the last four releases, i.e. one year's worth of spack-stack releases. Under special circumstances (e.g. operational implementations that require long-term support, long-running official retrospectives), selected releases may be kept longer. Maintenance in this context referts to addressing bugs or recompiling in the event of an intrusive update to the system. Adding new functionality or new packages is in general limited to the current and future releases.
```
Comment (climbfuji) [inline project_charter.md]: add sth like exceptions can be made on a case by case basis?

Comment (AlexanderRichert-NOAA) [reply project_charter.md]: :) see my comment

**Comment by climbfuji** [comment]
> What about an additional sentence: "Modifications to past releases that do not impact reinstallation, such as the addition of chained environments/custom Spack repositories, or new templates or site configurations, may be considered on a case-by-case basis."? Then in practice, we put as much of the development burden on the requesting developers as possible.
> 
> Also, referts->refers

that sounds much better than what I would have written ;-)

File: project_charter.md (hunk 1 above)
Comment (climbfuji) [inline project_charter.md]:
```suggestion
spack-stack developers are committed to maintaining a rolling stock of the last four releases, i.e. one year's worth of spack-stack releases. Under special circumstances (e.g. operational implementations that require long-term support, long-running official retrospectives), selected releases may be kept longer. Maintenance in this context refers to addressing bugs or recompiling in the event of an intrusive update to the system. Adding new functionality or new packages is in general limited to the current and future releases. Modifications to past releases that do not impact existing installations, such as the addition of chained environments/custom Spack repositories, or new templates or site configurations, may be considered on a case-by-case basis."
```

**Review by AlexanderRichert-NOAA — APPROVED** [review]

**Review by edwardhartnett — APPROVED** [review]

**Review by RatkoVasic-NOAA — APPROVED** [review]

**Pull request merged**